iml-manager-cli/src/stratagem.rs

```diff
@@ -323,21 +332,65 @@ pub async fn stratagem_cli(command: StratagemCommand) -> Result<(), ImlManagerCl
 
             wait_for_cmd_display(command).await?;
         }
-        StratagemCommand::Filesync(data) => {
-            let query = stratagem_queries::filesync::build(
-                &data.filesystem,
-                data.remote,
-                data.expression,
-                "push",
-            );
-
-            let resp: iml_graphql_queries::Response<stratagem_queries::filesync::Resp> =
-                graphql(query).await?;
+        StratagemCommand::Filesync(data) => match data.expression {
+            Some(ref _x) => {
+                let query = stratagem_queries::filesync::build(
+                    &data.filesystem,
+                    data.remote,
+                    data.expression.unwrap(),
+                    data.action,
+                );
+
+                let resp: iml_graphql_queries::Response<stratagem_queries::filesync::Resp> =
+                    graphql(query).await?;
 
-            let command = Result::from(resp)?.data.stratagem.run_filesync;
+                let command = Result::from(resp)?.data.stratagem.run_filesync;
 
-            wait_for_cmd_display(command).await?;
-        }
+                wait_for_cmd_display(command).await?;
+            }
+            None => {
+                #[derive(Serialize, Debug)]
+                struct FilesyncArgs {
+                    remote: String,
+                    action: String,
+                }
+
+                let task_args = FilesyncArgs {
+                    remote: data.remote,
+                    action: data.action.to_string(),
+                };
+
+                let llapi = match LlapiFid::create(&data.filesystem) {
+                    Ok(x) => x,
+                    Err(e) => {
+                        eprintln!("Can only specify a list of files on a client where the filesystem is mounted: {}", e);
+                        return Ok(());
+                    }
+                };
+
+                let (fids, errors): (Vec<_>, Vec<_>) = data
+                    .files
+                    .into_iter()
+                    .map(|file| llapi.path2fid(&file))
+                    .partition(Result::is_ok);
+                let fidlist: Vec<_> = fids.into_iter().map(Result::unwrap).collect();
+                let errors: Vec<_> = errors.into_iter().map(Result::unwrap_err).collect();
+                if !errors.is_empty() {
+                    eprintln!("files not found, ignoring: {:?}", errors);
+                }
+
+                let query = stratagem_queries::task_fidlist::build(
+                    "filesync",
+                    "filesync",
+                    &data.filesystem,
+                    serde_json::to_value(task_args).unwrap(),
```

Comment:
Avoid usage of `.unwrap()`

Reply:
I think this one is OK as well.

Reply:
This one should be simple to update:

```suggestion
                    serde_json::to_value(task_args)?,
```

Reply:
Done